app/models/task.py

```diff
@@ -1,5 +1,37 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy import Integer, String, Text, DateTime, ForeignKey
 from ..db import db
 
 class Task(db.Model):
-    id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    __tablename__ = "tasks"
+
+    id:           Mapped[int]       = mapped_column(Integer, primary_key=True, autoincrement=True)
+    title:        Mapped[str]       = mapped_column(String, nullable=False)
+    description:  Mapped[str]       = mapped_column(Text, nullable=False)
+    completed_at: Mapped[DateTime]  = mapped_column(DateTime, nullable=True)
+
+    # ← NEW: FK to goals.id
+    goal_id:      Mapped[int|None]  = mapped_column(
+                           ForeignKey("goals.id"),
+                           nullable=True
+    )
```

Comment:
@NataliaRaz How does `Mapped[int|None]` work? How would you explain this piece of code and this specific syntax used here `[something|something]`?

Reply:
Since SQLAlchemy is able to infer that this is a nullable field with `[X|None]` syntax, you **do not** also need to include `nullable=True`. Where possible we should avoid adding in unnecessary code because it increases the amount of code we have to maintain in  a codebase. 
```suggestion
    goal_id: Mapped[int|None] = mapped_column(ForeignKey("goals.id"))
```